libs/packages/components/src/lib/popover/popover.directive.ts

```diff
@@ -0,0 +1,104 @@
+import { AfterViewInit, Directive, ElementRef, HostListener, Input, Renderer2, TemplateRef } from '@angular/core';
+import { debounce } from './debounce.decorator';
+
+@Directive({
+  selector: '[sdsPopover]',
+  exportAs: 'sdsPopover'
+})
+export class SdsPopoverDirective implements AfterViewInit {
+  private _sdsPopoverContent: string | TemplateRef<any> | HTMLParagraphElement;
+  private _sdsPopoverTitle: string | TemplateRef<any> | HTMLParagraphElement;
+
+  sdsPopoverDiv: HTMLElement;
+  clicked= false;
+  popoverDivId: string;
+
+  @HostListener('click', ['$event']) onClick(){
+    this.clickEvent();
+  }
+
+  /**
+   * Adding listener for keyup.enter to ensure that user can activate popover with keyboard
+   */
+  @HostListener('keyup.enter', ['$event']) onKeyUp(){
+    this.clickEvent();
+  }
+
+  @Input()
+  position: string = 'top';
+
+  constructor(private el: ElementRef, private renderer: Renderer2) {
+    this.renderer.addClass(this.el.nativeElement, 'sds-popover');
+
+    this.sdsPopoverDiv = document.createElement('div');
+    this.renderer.addClass(this.sdsPopoverDiv, 'sds-popover__content');
+    this.renderer.addClass(this.sdsPopoverDiv, 'tooltip')
+  }
+
+  ngAfterViewInit() {
+    this.popoverDivId = this.el.nativeElement.id ? `${this.el.nativeElement.id}-popover` : `${this.el.nativeElement.tagName}-${this.el.nativeElement.offsetTop}-${this.el.nativeElement.offsetWidth}-popover`
+    this.sdsPopoverDiv.id = this.popoverDivId;
+    this.renderer.setAttribute(this.sdsPopoverDiv, 'data-position', this.position)
+    this.renderer.setAttribute(this.sdsPopoverDiv, 'aria-hidden', 'true');
+    this.renderer.setAttribute(this.el.nativeElement, 'role', 'button')
```

Comment:
In terms of accessibility for pop-overs, would it make sense to have aria-expanded and a has-popup attribute as well? - Something like this - https://codepen.io/esaramago/pen/jbXZaz
It looks like aria-expanded is a recommended attribute for popovers - https://techservicesillinois.github.io/accessibility/examples/popover.html#:~:text=Accessibility%20Examples%3A%20Popover%20Widget,(tooltips%20are%20not%20focusable).